src/pkg/services/m365/api/lists.go

```diff
@@ -62,3 +63,255 @@ func (c Lists) PostDrive(
 
 	return drive, graph.Wrap(ctx, err, "fetching created documentLibrary").OrNil()
 }
+
+// SharePoint lists represent lists on a site. Inherits additional properties from
+// baseItem: https://learn.microsoft.com/en-us/graph/api/resources/baseitem?view=graph-rest-1.0
+// The full details concerning SharePoint Lists can
+// be found at: https://learn.microsoft.com/en-us/graph/api/resources/list?view=graph-rest-1.0
+// Note additional calls are required for the relationships that exist outside of the object properties.
+
+// GetListById is a utility function to populate a SharePoint.List with objects associated with a given siteID.
+// @param siteID the M365 ID that represents the SharePoint Site
+// Makes additional calls to retrieve the following relationships:
+// - Columns
+// - ContentTypes
+// - List Items
+func (c Lists) GetListByID(ctx context.Context, siteID, listID string) (models.Listable, error) {
+	list, err := c.Stable.
+		Client().
+		Sites().
+		BySiteId(siteID).
+		Lists().
+		ByListId(listID).
+		Get(ctx, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "fetching list")
+	}
+
+	cols, cTypes, lItems, err := c.getListContents(ctx, siteID, listID)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "getting list contents")
+	}
+
+	list.SetColumns(cols)
+	list.SetContentTypes(cTypes)
+	list.SetItems(lItems)
+
+	return list, nil
+}
+
+// getListContents utility function to retrieve associated M365 relationships
+// which are not included with the standard List query:
+// - Columns, ContentTypes, ListItems
+func (c Lists) getListContents(ctx context.Context, siteID, listID string) (
+	[]models.ColumnDefinitionable,
+	[]models.ContentTypeable,
+	[]models.ListItemable,
+	error,
+) {
+	cols, err := c.getColumns(ctx, siteID, listID, "")
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	cTypes, err := c.getContentTypes(ctx, siteID, listID)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	lItems, err := c.getListItems(ctx, siteID, listID)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return cols, cTypes, lItems, nil
+}
+
+// getListItems utility for retrieving ListItem data and the associated relationship
+// data. Additional call append data to the tracked items, and do not create additional collections.
+// Additional Call:
+// * Fields
+func (c Lists) getListItems(ctx context.Context, siteID, listID string) ([]models.ListItemable, error) {
```

Comment:
Can you please asdd TODO comments for all `get*Items` funcs that they need to be transferred to the current pager implementation?